test/fuzz/luaL_loadbuffer/serializer.cc

```diff
@@ -184,16 +184,99 @@ GetCondition(const std::string &counter_name, const std::string &then_block)
 	return retval;
 }
 
+/**
+ * Class that registers and provides context during code
+ * generation.
+ * Used to generate correct Lua code.
+ */
+class Context {
+public:
+	enum BlockType {
+		kReturnable,
+		kBreakable,
+	};
+
+	void step_in(BlockType type)
+	{
+		block_stack_.push(type);
+		if (type == BlockType::kReturnable) {
+			++returnable_counter_;
+		}
+	}
+
+	void step_out()
+	{
+		assert(!block_stack_.empty());
+		if (block_stack_.top() == BlockType::kReturnable) {
+			assert(returnable_counter_ > 0);
+			--returnable_counter_;
+		}
+		block_stack_.pop();
+	}
+
+	std::string get_next_block_setup()
+	{
+		std::size_t id = GetCounterIdProvider().next();
+		std::string counter_name = GetCounterName(id);
+
+		return GetCondition(counter_name, get_exit_statement_()) +
+		       GetCounterIncrement(counter_name);
+	}
+
+	bool break_is_possible()
+	{
+		return !block_stack_.empty() &&
+		       block_stack_.top() == BlockType::kBreakable;
+	}
+
+	bool return_is_possible()
+	{
+		return returnable_counter_ > 0;
+	}
+
+private:
+
+	std::string get_exit_statement_()
+	{
+		assert(!block_stack_.empty());
+		switch (block_stack_.top()) {
+		case BlockType::kBreakable:
+			return "break";
+		case BlockType::kReturnable:
+			return "return";
+		}
```

Comment:
`default` case is missing.

Reply:
It is a switch on an enumerated value, so a default case was skipped by choice. The compiler will produce a warning if any values are not handled.

Reply:
An integer can be converted to any enum in C++, so if you want to protect this switch statement, I would prefer to add unreachable() from[ src/trivia/util.h](https://github.com/tarantool/tarantool/blob/06be32fbbdb884e407a923615606a3c0db599453/src/trivia/util.h#L260-L264) to keep compiler warnings for the future.

Reply:
We may add `unreachable()` *and* `assert(0)` to the `default` statement to be sure that this will be covered in the future.

Reply:
If some new enumerators will be added, and this switch won't be expanded, this mistake will be noticed only in a runtime. So I want  `unreachable()` / `assert(0)` after the switch statement.

Reply:
OK, I get it now. Thanks for the explanation. LGTM.